index.ts

```diff
@@ -136,22 +110,12 @@ export const sendBatchToS3 = async (payload: UploadJobPayload, meta: PluginMeta<
         params.SSEKMSKeyId = config.sseKmsKeyId
     }
 
-    console.log(`Flushing ${batch.length} events!`)
+    console.log(`Flushing ${events.length} events!`)
     global.s3.upload(params, async (err: Error, _: ManagedUpload.SendData) => {
         if (err) {
             console.error(`Error uploading to S3: ${err.message}`)
-            if (payload.retriesPerformedSoFar >= 15) {
-                return
-            }
-            const nextRetryMs = 2 ** payload.retriesPerformedSoFar * 3000
-            console.log(`Enqueued batch ${payload.batchId} for retry in ${nextRetryMs}ms`)
-            await jobs
-                .uploadBatchToS3({
-                    ...payload,
-                    retriesPerformedSoFar: payload.retriesPerformedSoFar + 1,
-                })
-                .runIn(nextRetryMs, 'milliseconds')
+            throw new RetryError()
```

Comment:
This would be an uncaught error as-is right and not be retried? Perhaps we should create a custom promise in sendBatchToS3 and return that?